maxim/tests/test_connection_retry_logic.py

```
@@ -464,6 +464,189 @@ def test_version_check_during_retry(self, mock_sleep, mock_scribe):
+                        mock_client
+                    )
+                    mock_client.request.side_effect = error
+
+                    # Test that each httpx error type is caught and handled
+                    with self.assertRaises(Exception):
+                        self.maxim_api.get_prompt("test-prompt-id")
+
+    @patch("maxim.apis.maxim_apis.scribe")
+    @patch("time.sleep")
+    def test_connect_timeout_with_exponential_backoff(self, mock_sleep, mock_scribe):
+        """Test that ConnectTimeout errors use the same exponential backoff as other connection errors."""
```

Pylint (3.3.7)

[convention] Line too long (106/100)

(C0301)